demos/text_detection_demo/src/text_recognition.cpp

```diff
@@ -53,3 +76,83 @@ std::string CTCGreedyDecoder(const std::vector<float> &data, const std::string&
     }
     return res;
 }
+
+std::string CTCBeamSearchDecoder(const std::vector<float> &data, const std::string& alphabet, char pad_symbol, double *conf, int bandwidth) {
+    std::string res = "";
+
+    const int num_classes = alphabet.length();
+
+    std::vector<BeamElement> curr;
+    std::vector<BeamElement> last;
+
+    last.push_back(BeamElement{"", 1.f, 0.f});
+
+    for (std::vector<float>::const_iterator it = data.begin(); it != data.end(); it += num_classes) {
+        curr.clear();
+
+        std::vector<float> prob = std::vector<float>(num_classes, 0.f);
+        softmax(it, it + num_classes, prob);
+
+        for (int candidate_num = 0; candidate_num < static_cast<int>(last.size()); candidate_num++) {
+            float prob_not_blank = 0.f;
+            auto candidate = last[candidate_num];
+            std::string candidate_sentence = candidate.sentence;
+            if (candidate_sentence != "") {
```

Comment:
If `sentence` field is `vector<int>`, then this line should be replaced with `if (!candidate_sentence.empty())`